torch/csrc/cuda/nccl.h

```diff
@@ -1,5 +1,7 @@
 #pragma once
 
+#include <nccl.h>
+
```

Comment:
iiuc we were including nccl.h later than defining these macros, that's bad.

Could you add a comment at the top of this file defining the contract, everyone in torch should include `torch/csrc/cuda/nccl.h` and nobody should includ `nccl.h` directly, and all NCCL Version checking should be done in this file, iiuc?

Reply:
Yes, let's add a note here!